subprojects/docs/src/docs/release/notes.md

```diff
@@ -58,6 +58,14 @@ In this way, stale execution data, which would cause non-repeatable task outputs
 Since Gradle now takes care of removing the execution data, the `JacocoPluginExtension.append` property has been deprecated.
 The JaCoCo agent is always configured with `append = true`, so it can be used when running tests in parallel. 
 
+### Plural task output properties don't disable caching anymore
```

Comment:
I think we also need to add something/change the user guide and adjust the Javadocs on `OutputFiles` and `OutputDirectories`.